vscode-trace-extension/src/utils/trace-extension-logger.ts

```diff
@@ -0,0 +1,186 @@
+/***************************************************************************************
+ * Copyright (c) 2023 BlackBerry Limited and others.
+ *
+ * Licensed under the MIT license. See LICENSE file in the project root for details.
+ ***************************************************************************************/
+
+import * as vscode from 'vscode';
+
+export class TraceExtensionLogger {
+    private logChannel: vscode.OutputChannel;
+    private channelDisposed = false;
+
+    constructor(logChannelName: string) {
+        this.logChannel = vscode.window.createOutputChannel(logChannelName);
+    }
+
+    /**
+     * Returns the status of the log output channel
+     *
+     * @returns true if the current channel has been disposed, false otherwise
+     */
+    isCurrentChannelDisposed(): boolean {
+        return this.channelDisposed;
+    }
+
+    /**
+     * Show the log channel from the UI
+     */
+    showLog(): void {
+        if (!this.channelDisposed) {
+            this.logChannel.show();
+        }
+    }
+
+    /**
+     * Hide the log channel from the UI
+     */
+    hideLog(): void {
+        if (!this.channelDisposed) {
+            this.logChannel.hide();
+        }
+    }
+
+    /**
+     * Clears the entries from the log channel
+     */
+    clearLog(): void {
+        if (!this.channelDisposed) {
+            this.logChannel.clear();
+        }
+    }
+
+    /**
+     * Disposes the log channel and all of its resources
+     */
+    disposeChannel(): void {
+        this.logChannel.dispose();
```

Comment:
If one calls this method more than once in a row, then some `dispose()` implementation might eventually break. Should we check for `isCurrentChannelDisposed()` before calling this here?